packages/cozy-client/src/CozyClient.spec.js

```diff
@@ -1402,6 +1401,42 @@ describe('CozyClient', () => {
         receiveQueryResult('allTodos', fakeResponse)
       )
     })
+
+    const setupOnError = () => {
+      const error = new TypeError('Failed to fetch')
+      requestHandler.mockReturnValueOnce(Promise.reject(error))
+      const onError = jest.fn().mockReturnValueOnce(() => true)
+      return onError
+    }
+
+    it('should call onError callback when an error is catched. It is passed as argument', async () => {
+      const onError = setupOnError()
+
+      try {
+        await client.query(query, { as: 'allTodos', onError })
+      } catch (e) {} // eslint-disable-line no-empty
+      expect(onError).toBeCalled()
+    })
+
+    it('should call onError callback when an error is catched. It is passed during instantiation', async () => {
+      const onError = setupOnError()
+      let client = new CozyClient({ onError })
+
+      try {
+        await client.query(query, { as: 'allTodos' })
+      } catch (e) {} // eslint-disable-line no-empty
+      expect(onError).toBeCalled()
+    })
+
+    it('should call onError callback when an error is catched. It is passed with setOnError', async () => {
+      const onError = setupOnError()
+      client.setOnError(onError)
+
+      try {
+        await client.query(query, { as: 'allTodos' })
+      } catch (e) {} // eslint-disable-line no-empty
+      expect(onError).toBeCalled()
+    })
```

Comment:
We should remove the try / catch block when there is an `onError` callback. 

But we should add a test to test that if we don't have any onError callback the error is thrown and catchable.